libs/api/ebpf_api.cpp

```diff
@@ -377,6 +407,56 @@ _get_program_information_data(ebpf_program_type_t program_type, ebpf_extension_d
     return retval;
 }
 
+uint32_t
+ebpf_get_program_byte_code(
```

Comment:
Shouldn't there be a public header for this function?

Reply:
My understanding was that the public APIs will be the one which will be exposed to the user mode app? If that is the case, then this API will not be exposed to user mode app, but internally used by ebpfclient. Hence I did not add it to the ebpf_api.h.

Reply:
If it's not used outside this CPP file, it should be static.
If it's used outside this CPP file, it should be declared in a header.

Having two versions of this function (one in main.cpp and one in ebpf_api.cpp) can lead to problems if they get out of sync. Better to have a single common header to avoid [ODR](https://en.wikipedia.org/wiki/One_Definition_Rule) violations.

Reply:
By "versions of this function", do you mean declarations? there is only one definition for this function. I will add it to the public header.

Reply:
The problem is if there isn't a common header, then you run the risk of the signatures getting out of synch, which can result in hard to debug failures.

Reply:
Added it to a new header file api_internal.h